evm/trie/trie_test.go

```diff
@@ -0,0 +1,455 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package trie
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethdb/leveldb"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+var (
+	defaultTriesToStore = 3
+)
+
+func createNewTxTries(numHistoricalTries int) *TxTries {
+	return NewTxTries(numHistoricalTries)
+}
+
+func createTempDB() *leveldb.Database {
+
+	diskdb, err := leveldb.New("./temp-database", 256, 0, "")
+	if err != nil {
+		panic(fmt.Sprintf("unable to create testing database: %v", err))
+	}
+	return diskdb
+}
+
+func deleteTempDB() {
+	os.RemoveAll("./temp-database")
+}
```

Comment:
To avoid having test artifacts lying around and to ensure they are never reused between tests, it would be better to use a temporary directory. Something like `ioutil.TempDir(os.TempDr(), "temp-db-*")` will create a directory in your systems default temp directory (`os.TempDir()`) with the name "temp-db-somerandomstring".

To cleanup, it might be easiest to return a function from `createTempDB` that calls `os.Remove` on the temo dir. This is not as important with temp dirs, but still good practice to cleanup.